### PR TITLE
ISPN-15824 Collect tls endpoint from protocol servers

### PR DESCRIPTION
**Pull request by fax4ever**
https://issues.redhat.com/browse/ISPN-15823
https://issues.redhat.com/browse/ISPN-15824

### CODE REVIEW

File: server/runtime/src/main/java/org/infinispan/server/Server.java [hunk 1]
```diff
@@ -810,19 +810,21 @@ public Json securityOverviewReport() {
       Json securityRealms = Json.object();
       for (Map.Entry<String, RealmConfiguration> realm : serverConfiguration.security().realms().realms().entrySet()) {
          RealmConfiguration realConfig = realm.getValue();
-         securityRealms.set(realm.getKey(), Json.object(
-               "server-ssl", realConfig.hasServerSSLContext(), "client-ssl", realConfig.hasClientSSLContext()));
+         if (realConfig.hasClientSSLContext()) {
```
Comment (tristantarrant) [inline server/runtime/src/main/java/org/infinispan/server/Server.java:813]: This is wrong. Ignore the client context.
```java
        if (realmConfig.hasServerSSLContext()) {
            if (realmConfig.hasFeature(ServerSecurityRealm.Feature.TRUST)) {
               securityRealms.set(realm.getKey(), Json.object("tls", "CLIENT"));
            } else {
               securityRealms.set(realm.getKey(), Json.object("tls", "SERVER"));
            }
         } else {
            securityRealms.set(realm.getKey(), Json.object("tls", "NONE"));
         }
```

**Review by tristantarrant — CHANGES_REQUESTED** [review]

**Comment by fax4ever** [comment]
Pushed the change, thanks @tristantarrant

**Comment by tristantarrant** [comment]
I would like to see some tests in `AuthenticationTLSIT` (should return `CLIENT`) `TLSWithoutAuthenticationIT` (should return `SERVER`) and `OverviewReportIT` (should return `NONE`).

**Comment by tristantarrant** [comment]
Merged, thanks

**Comment by fax4ever** [comment]
Thank you @tristantarrant

**Pull request merged**